modules/api_test/tests/src/Functional/ApiTest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace Drupal\Tests\api_test\Functional;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Tests\BrowserTestBase;
+
+/**
+ * @group headless
+ * @group api_test
+ */
+class ApiTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $profile = 'lightning_headless';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = ['api_test'];
+
+  /**
+   * The access token returned by the api.
+   *
+   * @var string
+   */
+  protected $accessToken;
+
+  /**
+   * URL strings for different endpoints.
+   *
+   * @var string[]
+   */
+  protected $paths = [
```

Comment:
For clarity only. I found myself having trouble parsing what the URLs meant when they were hard-coded.